### ISSUE TITLE
Java "resteasy" ApiClient not adding default HTTP headers correctly

### ISSUE THREAD

**Issue by paulrene** [swagger-api/swagger-codegen#7758]
##### Description
In the ApiClient.java file around line 592 the following code is responsible for adding the default http headers to the request.

    for (Entry<String, String> defaultHeaderEnrty: defaultHeaderMap.entrySet()) {
      if (!headerParams.containsKey(defaultHeaderEnrty.getKey())) {
        String value = defaultHeaderEnrty.getKey();
        if (value != null) {
          invocationBuilder = invocationBuilder.header(defaultHeaderEnrty.getKey(), value);
        }
      }
    }

The line "String value = defaultHeaderEnrty.getKey();" is wrong and should have been "String value = defaultHeaderEnrty.getValue();"

##### Swagger-codegen version
2.3.1

**Comment by jmini** [comment]
Correct me if I am wrong, but I am trying to follow your issue…
As far as I know, the `ApiClient.java` you mention is generated by the `swagger-codegen` generator. Nobody else has the same code. Line 592 is not something that can be used to figure out your problem. 

In which method is your problem? Is it in `ApiClient.invokeAPI(..)`

I guess you have used the `java` lang as parameter to generate your code. Any other options set? I guess you are using the `resteasy` library.
Can you confirm?

---

Your problem might be located in this template: [`/modules/swagger-codegen/src/main/resources/Java/libraries/resteasy/ApiClient.mustache`](https://github.com/jmini/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/Java/libraries/resteasy/ApiClient.mustache) around line L599

If you compare it with the [jersey2 version](https://github.com/jmini/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/Java/libraries/jersey2/ApiClient.mustache), the for loop to handle `defaultHeaderMap` looks like this:

```
for (Entry<String, String> entry : defaultHeaderMap.entrySet()) {
      String key = entry.getKey();
      if (!headerParams.containsKey(key)) {
        String value = entry.getValue();
        if (value != null) {
          invocationBuilder = invocationBuilder.header(key, value);
        }
      }
}
```

You should see if this solves your problem and then push a pull request. If something is not clear or if you need help, please continue the discussion.

**Comment by paulrene** [comment]
You are correct. The bug is located here: https://github.com/jmini/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/Java/libraries/resteasy/ApiClient.mustache#L601

defaultHeaderEnrty.getKey() should have been defaultHeaderEnrty.getValue()

**Comment by jmini** [comment]
Yes and `defaultHeaderEnrty` could be renamed `entry` or at least `defaultHeaderEntry`.

Do you want to file a pull request for that?

**Comment by paulrene** [comment]
Creating pull request.

**Comment by thebeeber** [comment]
Thanks for catching and fixing this.  Is there a schedule for when this will make it into the next release?  We really need this for a project we are working on.  Thanks!
